uniffi_bindgen/src/bindings/python/templates/ErrorTemplate.py

```diff
@@ -1,19 +1,23 @@
 {%- let e = ci.get_error_definition(name).unwrap() %}
-class {{ type_name }}(Exception):
-    {%- if e.is_flat() %}
 
-    # Each variant is a nested class of the error itself.
-    # It just carries a string error message, so no special implementation is necessary.
-    {%- for variant in e.variants() %}
-    class {{ variant.name()|class_name }}(Exception):
+# {{ type_name }}
+# We want to define each variant as a nested class that's also a subclass,
+# which is tricky in Python.  To accomplish this we're going to create each
+# class separated, then manually add the child classes to the base class's
+# __dict__.  All of this happens in dummy class to avoid polluting the module
+# namespace.
+class UniFFIExceptionTmpNamespace:
+    class {{ type_name }}(Exception):
         pass
-    {%- endfor %}
+    {% for variant in e.variants() %}
+    {%- let variant_type_name = variant.name()|class_name %}
 
+    {%- if e.is_flat() %}
+    class {{ variant_type_name }}({{ type_name }}):
+        def __unicode__(self):
```

Comment:
I thought this was `__str__` in python 3? https://docs.python.org/3/reference/datamodel.html#object.__str__ doesn't mention `__unicode__` and a quick test:

```
class T:
#  def __str__(self):
#    return "str"
  def __unicode__(self):
    return "uni"

t = T()
print(t)
```

As shown, that prints `<__main__.T object at 0x00000156BC9CA140>` but if `__str__` is uncommented, `str` is printed